packages/salesforcedx-vscode-automation-tests/test/specs/debugLwcTests.e2e.ts

```diff
@@ -0,0 +1,271 @@
+    await lwcTestsItems[0].select();
+    const debugTestsAction = await lwcTestsItems[0].getActionButton('SFDX: Debug Lightning Web Component Test File');
+    expect(debugTestsAction).to.not.be.undefined;
+    await debugTestsAction!.click();
+    await pause(Duration.seconds(15));
+
+    // Continue with the debug session
+    await continueDebugging(2);
+
+    // Verify test results are listed on the terminal
+    // Also verify that all tests pass
+    const terminalText = await getTerminalViewText(workbench, 10);
+    const expectedTexts = [
+      'PASS  force-app/main/default/lwc/lwc1/__tests__/lwc1.test.js',
+      'Test Suites: 1 passed, 1 total',
+      'Tests:       2 passed, 2 total',
+      'Snapshots:   0 total',
+      'Ran all test suites within paths',
+      `${path.join('force-app', 'main', 'default', 'lwc', 'lwc1', '__tests__', 'lwc1.test.js')}`
+    ];
+    expect(terminalText).to.not.be.undefined;
+    await verifyOutputPanelText(terminalText!, expectedTexts);
+
+    // Verify the tests that are passing are labeled with a green dot on the Test sidebar
+    await executeQuickPick('Testing: Focus on LWC Tests View', Duration.seconds(3));
+    await verifyTestIconColor(lwcTestsItems[0], 'testPass');
+  });
+
+  step('Debug Single Test via the Test Sidebar', async () => {
+    log(`${testSetup.testSuiteSuffixName} - Debug Single Test via the Test Sidebar`);
+    const workbench = getWorkbench();
+    const testingView = await workbench.getActivityBar().getViewControl('Testing');
+    expect(testingView).to.not.be.undefined;
+    // Open the Test Sidebar
+    const testingSideBarView = await testingView?.openView();
+    expect(testingSideBarView).to.be.instanceOf(SideBarView);
+
+    // Hover a test name under one of the test lwc sections and click the debug button that is shown to the right of the test name on the Test sidebar
+    const lwcTestsSection = await getTestsSection(workbench, 'LWC Tests');
+    const lwcTestItem = (await lwcTestsSection.findItem('displays greeting')) as TreeItem;
+    await lwcTestItem.select();
+    const debugTestAction = await lwcTestItem.getActionButton('SFDX: Debug Lightning Web Component Test Case');
+    expect(debugTestAction).to.not.be.undefined;
+    await debugTestAction!.click();
+    await pause(Duration.seconds(15));
+
+    // Continue with the debug session
+    await continueDebugging(1);
+
+    // Verify test results are listed on the terminal
+    // Also verify that all tests pass
+    const terminalText = await getTerminalViewText(workbench, 10);
+    const expectedTexts = [
+      'PASS  force-app/main/default/lwc/lwc1/__tests__/lwc1.test.js',
+      'Test Suites: 1 passed, 1 total',
+      'Tests:       1 skipped, 1 passed, 2 total',
+      'Snapshots:   0 total',
+      'Ran all test suites within paths',
+      `${path.join('force-app', 'main', 'default', 'lwc', 'lwc1', '__tests__', 'lwc1.test.js')}`
+    ];
```

Comment:
I wonder how hard it would be to use `jest` for these to be able to get snapshots?

SDR used `mocha-snap` if mocha is required for framework reasons https://github.com/forcedotcom/source-deploy-retrieve/tree/main/test/snapshot

I also like that better than big strings as exepctations